tests/queries/tests.py

```diff
@@ -2016,6 +2016,17 @@ def test_ordered_subselect(self):
         query = DumbCategory.objects.filter(id__in=DumbCategory.objects.order_by('-id')[2:])
         self.assertEqual(set(query.values_list('id', flat=True)), {1, 2})
 
+    def test_ordered_select_for_update(self):
+        """
+        Subqueries containing select_for_update should respect ordering. Users
```

Comment:
"should respect ordering as an ORDER BY clause may be useful to specify a row locking order to prevent deadlocks (#27193)."